electron_app/src/electron-main.js

```diff
@@ -186,6 +197,66 @@ app.on('ready', () => {
         }
     }
 
+    protocol.registerFileProtocol('vector', (request, callback) => {
+        if (request.method !== 'GET') {
+            callback({error: -322}); // METHOD_NOT_SUPPORTED from chromium/src/net/base/net_error_list.h
+            return null;
+        }
+
+        const parsedUrl = new URL(request.url);
+        if (parsedUrl.protocol !== 'vector:') {
+            callback({error: -302}); // UNKNOWN_URL_SCHEME
+            return;
+        }
+        if (parsedUrl.host !== 'vector') {
+            callback({error: -105}); // NAME_NOT_RESOLVED
+            return;
+        }
+
+        const target = parsedUrl.pathname.split('/');
+
+        // path starts with a '/'
+        if (target[0] !== '') {
+            callback({error: -6}); // FILE_NOT_FOUND
+            return;
+        }
+
+        if (target[target.length - 1] == '') {
+            target[target.length - 1] = 'index.html';
+        }
+
+        let baseDir;
+        // first part of the path determines where we serve from
+        if (target[1] === 'origin_migrator_dest') {
+            // the origin migrator destination page
+            // (only the destination script needs to come from the
+            // custom protocol: the source part is loaded from a
+            // file:// as that's the origin we're migrating from).
+            baseDir = __dirname + "/../../origin_migrator/dest";
+        } else if (target[1] === 'webapp') {
+            baseDir = __dirname + "/../../webapp";
```

Comment:
might be better to use `path.join` for this and the one above, because Windows. Not sure how badly it'll complain though.

Reply:
nvm, `path.normalize` fixes it all